crates/sui-core/src/authority.rs

```diff
@@ -569,6 +599,54 @@ impl AuthorityState {
         &self.committee_store
     }
 
+    pub fn get_google_jwk_as_bytes(&self) -> Vec<u8> {
+        match self.google_jwk_as_bytes.read() {
+            Ok(google_jwk_as_bytes) => google_jwk_as_bytes.clone(),
+            Err(_) => (*DEFAULT_GOOGLE_JWK_BYTES.clone()).to_vec(),
+        }
+    }
+
+    pub fn start_jwk_updater(self: Arc<Self>) {
+        debug!("Starting JWK updater thread for authority");
+        tokio::task::spawn(async move {
+            loop {
+                // Update the JWK value in the authority server
+                let res = self.update_google_jwk().await;
+                if let Err(e) = res {
+                    debug!("Error when fetching JWK {:?}", e);
+                }
+                // Sleep for 1 hour
+                tokio::time::sleep(Duration::from_secs(3600)).await;
+            }
+        });
+    }
+
+    async fn update_google_jwk(&self) -> Result<(), SuiError> {
+        let client = Client::new();
+        let response = client
+            .get("https://www.googleapis.com/oauth2/v2/certs")
```

Comment:
i think you will probably have to bypass doing an actual http fetch in order for this to work with simtest. but it may not matter until you add specific zklogin tests.

Reply:
makes sense. i can just use a cfg(test) vs cfg(not(test)) for simtest to use a dummy response right?